.github/workflows/docker.yml

```diff
@@ -0,0 +1,43 @@
+name: Push Docker images to Docker Hub 
+
+on:
+  push:
+    branches: [main, master]
+
+jobs:
+  multiple-registries:
+    if: github.repository_owner == 'exercism' # Stops this job from running on forks.
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # 2.4.0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@94ab11c41e45d028884a99163086648e898eed25 # 1.6.0
+
+      - name: Cache Docker layers
+        uses: actions/cache@937d24475381cd9c75ae6db12cb4e79714b926ed # 2.1.7
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9 # 1.10.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@a66e35b9cbcf4ad0ea91ffcaf7bbad63ad9e0229 # 2.7.0
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: |
+            ${{ github.event.repository.full_name }}:latest
+            ${{ github.event.repository.full_name }}:${{ github.sha }}
```

Comment:
I assume you'll be using either latest or the sha when referring this Docker image? As in: no versions.

Reply:
To be honest, I just copied straight out of the `nim-test-runner` one. I don't think we'll be needing version unless different tools want to run different versions of nim. For now I don't think there is a need to version.

Do you think it's useful to keep the sha version?

Reply:
In this: not really, as we are in full control of the contents of the image.

Reply:
I think I'm gonna keep this as to be able to test for each tooling on each update of the base image and if something doesn't work it gives us the flexibility when needed.
It shouldn't pollute ECS or Docker Hub as there won't be many changes other optimization and/or nim upgrades.